### PR TITLE
test[next]: Add unit test for embedded `inverse_image` and fix bugs

### PR DESCRIPTION
**Pull request by egparedes**
Add unit test for `ConnectivityField.inverse_image()`.

### CODE REVIEW

File: tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py [hunk 1]
```diff
@@ -632,3 +632,59 @@ def test_setitem_wrong_domain():
...
+
+    # Test range
+    image_range = UnitRange(V_START, V_STOP)
+    result = e2v_conn.inverse_image(image_range)
+
+    assert len(result) == 1
+    assert result[0] == (E, UnitRange(V_START + 1, V_STOP + 1))
+
+    # Test cache
+    cached_result = e2v_conn.inverse_image(image_range)
```
Comment (havogt) [inline tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py:663]: why is this testing the cache?

File: tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py [hunk 2]
```diff
@@ -632,3 +632,59 @@ def test_setitem_wrong_domain():
...
+
+    assert result is cached_result
+
+    # Test codomain
+    with pytest.raises(ValueError, match="does not match the codomain dimension"):
+        e2v_conn.inverse_image((E, UnitRange(1, 2)))
+
+    # Test non-contiguous range
```
Comment (havogt) [inline tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py:671]: I'd say this should be a separate test.

File: tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py [hunk 3]
```diff
@@ -632,3 +632,59 @@ def test_setitem_wrong_domain():
 
     with pytest.raises(ValueError, match=r"Incompatible 'Domain'.*"):
         field[(1, slice(None))] = value_incompatible
+
+
```
Comment (havogt) [inline tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py:636]: maybe also add a testcase with a 2D connectivity?

Comment (havogt) [reply tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py:636]: To be honest I don't really understand if it makes sense to restrict the connectivity in that case. I have to think about it.

Comment (havogt) [reply tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py:636]: note to myself:

Start from this case and look at corner cases
```
E_START, E_STOP = 0, 1
V_START, V_STOP = 0, 1

e2v_conn = common._connectivity([0,1], domain=domain(named_range(E,1), named_range(E2V, 2)))
```

the inverse image is probably

```
domain(named_range(E,0), named_range(E2V, 1))
```

**Review by havogt — COMMENTED** [review]
I'd like to discuss the corner cases in the 2d case (also for understanding the extension to the one with skip values.

File: tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py (hunk 1 above)
Comment (egparedes) [reply tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py:663]: This is testing the cache because it returns exactly the same result object as before. If the cache were not used, it would return a new object with exactly the same content.

Comment (havogt) [reply tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py:663]: oh, I read the `is` as `==`

File: tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py (hunk 3 above)
Comment (egparedes) [reply tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py:636]: I've added some cases to test 2d connectivities

**Review by havogt — APPROVED** [review]
:+1:
